sdk/src/quotes/swap/tick-array-sequence.ts

```diff
@@ -3,32 +3,48 @@ import {
   MAX_TICK_INDEX,
   MIN_TICK_INDEX,
   TickArray,
+  TickArrayData,
   TickData,
   TICK_ARRAY_SIZE,
 } from "../../types/public";
 import { TickArrayIndex } from "./tick-array-index";
 import { PublicKey } from "@solana/web3.js";
 
+type InitializedTickArray = TickArray & {
+  // override
+  data: TickArrayData;
+};
+
 /**
  * NOTE: differs from contract method of having the swap manager keep track of array index.
  * This is due to the initial requirement to lazy load tick-arrays. This requirement is no longer necessary.
  */
 export class TickArraySequence {
+  private tickArrays: InitializedTickArray[];
   private touchedArrays: boolean[];
   private startArrayIndex: number;
 
   constructor(
-    readonly tickArrays: TickArray[],
+    tickArrays: TickArray[],
     readonly tickSpacing: number,
     readonly aToB: boolean
   ) {
     if (!tickArrays[0] || !tickArrays[0].data) {
       throw new Error("TickArray index 0 must be initialized");
     }
 
-    this.touchedArrays = [...Array<boolean>(tickArrays.length).fill(false)];
+    // If an uninitialized TickArray appears, truncate all TickArrays after it (inclusive).
+    this.tickArrays = [];
+    for (let i=0; i < tickArrays.length && !!tickArrays[i] && !!tickArrays[i].data; i++) {
```

Comment:
style nit ideas to get rid of the ! hard cast:

1. do the null check in the for loop closure instead of the for loop conditions. early break when we detect a null (most straightforward)
2. keep the current for loop but do the null check again so typescript recognizes it's not null. 
3. make a copy of input tickArray, do a reduce. splice the array when we get to a null to early exit a reduce (peak typescript ridiculous-ness)

Reply:
@odcheung 
Thanks for the comment & ideas.🙂
I have refactored the code using the 1st straightforward way.